src/main/java/uk/gov/companieshouse/api/accounts/model/entity/AccountDataEntity.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.companieshouse.api.accounts.model.entity;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import javax.validation.constraints.NotNull;
+import org.springframework.data.mongodb.core.mapping.Field;
+import uk.gov.companieshouse.GenerateEtagUtil;
+import uk.gov.companieshouse.api.accounts.Kind;
+import uk.gov.companieshouse.api.accounts.LinkType;
+
+public class AccountDataEntity {
+
+    @Field("id")
+    private String id;
+
+    @NotNull
+    @Field("period_end_on")
+    private LocalDate periodEndOn;
+
+    private Map<String, String> links = new HashMap<>();
+
+    private String kind;
+
+    private String etag;
+
+    protected AccountDataEntity build(String id){
+        this.id = id;
+        this.etag = GenerateEtagUtil.generateEtag();
+        this.kind = Kind.ACCOUNT.getValue();
+
+        Map<String, String> links = new HashMap<>();
+        
+        links.put(LinkType.SELF.getLink(), "");
+        this.links = links;
+        return this;
+    }
+
+    public String getId() {
```

Comment:
Can we not use something like Project Lombok to clean up the get/set bloat?

Reply:
Nothing wrong with that, however just to get this story done we could do it later on? I already have a few things I want to do , lombok was actually one of them, as well as using jetty and speeding up start time

Reply:
Fine with that, was just a "nice to have" I guess. Maybe we can get a story/spike drafted to look into it and bring it in soon.

Reply:
It can definitely be added to the list of todos